packages/ui/src/common/helpers.ts

```diff
@@ -53,3 +55,28 @@ export const subtitleMapping = (value: string) => {
       return value
   }
 }
+
+export const resizeImageFile = (file: File, width: number, height: number, type?: string): Promise<Blob | null> => {
+  return new Promise((resolve, reject) => {
+    if (!SUPPORTED_IMAGES.includes(file.type)) {
+      reject(new Error('Wrong file type'))
+      return
+    }
+
+    const reader = new FileReader()
+    reader.onload = (event) => {
+      const img = new Image()
+      img.onload = () => {
+        const canvas = document.createElement('canvas')
+        canvas.height = height
+        canvas.width = width
+        const ctx = canvas.getContext('2d')
+
+        ctx?.drawImage(img, 0, 0, width, height)
```

Comment:
Currently the image are ratio is modified. After some trial and error on: https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_canvas_drawimage
It looks like this should keep ratio (and center on the result image):
```suggestion
        const dW = img.width - width
        const dH = img.height - height
        const [clippedWidth, clippedHeight] = Math.abs(dW) > Math.abs(dH)
          ? [Math.floor(img.width / img.height * width), img.height]
          : [img.width, Math.floor(img.height / img.width * height)]

        const x = Math.floor(img.width/2 - clippedWidth/2)
        const y = Math.floor(img.height/2 - clippedHeight/2)

        ctx.drawImage(img, x, y, clippedWidth, clippedHeight, 0, 0, width, height);
```

Reply:
This basically depends on the image, bigger ones with context not in the center will seem meaningless when we keep the ratio and clip it.

Reply:
Yes but IMO (once one dimension of the image matches the target) it's preferable to crop it at an arbitrary position than deform it. Especially when the original ratio is kinda close to the targeted one, it makes me feel like something is not right :thinking: 

Also here the cropping happens in one dimension only.